src/Voron/Data/Tables/TableValueCompressor.cs

```diff
@@ -0,0 +1,419 @@
+                        }
+
+                        table.CurrentCompressionDictionaryId = newId;
+                        compressionDictionary.ExpectedCompressionRatio = GetCompressionRatio(CompressedBuffer.Length, RawBuffer.Length);
+
+                        var rev = Bits.SwapBytes(newId);
+                        using (Slice.External(tx.Allocator, (byte*)&rev, sizeof(int), out var slice))
+                        using (dictionariesTree.DirectAdd(slice, sizeof(CompressionDictionaryInfo) + dictionaryBufferSpan.Length, out var dest))
+                        {
+                            *((CompressionDictionaryInfo*)dest) =
+                                new CompressionDictionaryInfo {ExpectedCompressionRatio = compressionDictionary.ExpectedCompressionRatio};
+                            Memory.Copy(dest + sizeof(CompressionDictionaryInfo), dictionaryBuffer.Ptr, dictionaryBufferSpan.Length);
+                        }
+
+                        tx.LowLevelTransaction.OnDispose += RecreateRecoveryDictionaries;
+                    }
+                }
+            }
+            finally
+            {
+                ArrayPool<long>.Shared.Return(dataIds);
+                ArrayPool<UIntPtr>.Shared.Return(sizes);
+            }
+        }
+
+        public static readonly byte[] EncryptionContext = Encoding.UTF8.GetBytes("Compress");
+
+        private static void RecreateRecoveryDictionaries(IPagerLevelTransactionState obj)
+        {
+            if (!(obj is LowLevelTransaction llt) || llt.Committed == false)
+                return; // we can't write on non committed transactions
+
+            if (obj.Environment.Options is StorageEnvironmentOptions.PureMemoryStorageEnvironmentOptions)
+                return; // no need for in mem mode
+
+            lock (obj.Environment.CompressionDictionariesHolder)
+            {
+                using var tx = obj.Environment.ReadTransaction();
+
+                var dictionaries = tx.ReadTree(TableSchema.CompressionDictionariesSlice);
+
+                if (dictionaries == null)
+                {
+                    Debug.Assert(dictionaries != null);
+                    return; // should never happen
+                }
+
+                int nonceSize = (int)Sodium.crypto_stream_xchacha20_noncebytes();
+                var subKeyLen = (int)Sodium.crypto_aead_xchacha20poly1305_ietf_keybytes();
+                int macSize = (int)Sodium.crypto_aead_xchacha20poly1305_ietf_abytes();
+                var nonceBuffer = stackalloc byte[nonceSize];
+                var macBuffer = stackalloc byte[macSize];
+                var subKey = stackalloc byte[subKeyLen];
+
+                // for reliability's sake, we keep two copies of the compression
+                // dictionaries
+                for (int i = 0; i < 2; i++)
+                {
+                    var newPath = obj.Environment.Options.BasePath
+                        .Combine(path: $"Compression{(i == 0 ? "A" : "B")}.Recovery")
```

Comment:
Can we move `.Recovery` to some const and use it here and in the Recovery.cs ? Maybe also consider changing the extension to `.compression-recovery`? It is easier to recognize the files via extension than the name. Some day we might add more 'recovery' files.

Reply:
done